src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpExchange.java

```diff
@@ -199,9 +209,62 @@ protected HttpExchange() {
      *                       specified and no response body may be written.
      * @throws IOException   if the response headers have already been sent or an I/O error occurs
      * @see   HttpExchange#getResponseBody()
+     * @see   HttpExchange#CHUNKED_CONTENT
+     * @see   HttpExchange#NO_CONTENT
      */
     public abstract void sendResponseHeaders(int rCode, long responseLength) throws IOException;
 
+    /**
+     * Convenience method to send the response headers with no response body.
+     * The exchange is also closed.
+     * @param code the response code to send
+     * @throws IOException if the response headers have already been sent or an I/O error occurs
+     * @see HttpExchange#sendResponseHeaders(int, long)
+     */
+    public final void sendNoContentResponse(int code) throws IOException {
+        sendResponseHeaders(code,NO_CONTENT);
+    }
+    /**
+     * Convenience method to start sending a chunked response. The caller should write the response to
+     * the returned output stream, then close the output stream,
+     * @param code the response code to send
+     * @throws IOException if the response headers have already been sent or an I/O error occurs
+     * @return the stream to write the response to. the caller must close the stream.
+     * @see HttpExchange#sendResponseHeaders(int, long)
+     */
+    public final OutputStream beginChunkedResponse(int code) throws IOException {
+        sendResponseHeaders(code,CHUNKED_CONTENT);
+        return getResponseBody();
+    }
+    /**
+     * Convenience method to start sending a fixed length response. The caller should write the response to
+     * the returned output stream then close the stream. The number of bytes written to the stream must match
+     * the provided length.
+     * @param code the response code to send
+     * @param length the number of bytes that will be sent
+     * @throws IOException if the response headers have already been sent or an I/O error occurs
+     * @return the stream to write the response to. the caller must close the stream.
+     * @see HttpExchange#sendResponseHeaders(int, long)
+     */
+    public final OutputStream beginFixedLengthResponse(int code, long length) throws IOException {
+        sendResponseHeaders(code,length);
+        return getResponseBody();
+    }
```

Comment:
I'm not convinced that having this method will be enough to prevent common errors.

* It is more verbose than `sendResponseHeaders`
* As you've pointed out you need to have knowledge of the http protocol in order to grok what is going on. "Sending headers" might still be reached for first by gen-pop
* The `length` parameter here would be subtly different than the `responseLength` in `sendResponseHeaders`. (the 0, -1 thing). This feels like it makes the situation worse (not only is it weird to have -1 be no content and 0 be chunked - now its inconsistent)